### PR TITLE
CORS-3469: GCP IPI minimum permissions and without 'iam.serviceAccounts.actAs' testing

### PR DESCRIPTION
**Pull request by jianli-wei**
- cont. to https://github.com/openshift/release/pull/55925
- add non-shared-VPC testing, e.g. installing using minimum permissions and without 'iam.serviceAccounts.actAs' permission, along with specifying controlPlane.platform.gcp.serviceAccount
- combine the testing with existing marketplace tests, i.e. replace `gcp-ipi-marketplace` with `gcp-ipi-mini-perm-marketplace` in 4.17 & 4.18 amd64 nightly

FYI the installer SA is granted the Service Account User role on the control-plane SA: 
```
$ gcloud iam service-accounts get-iam-policy jiwei-control-plane-sa@openshift-qe.iam.gserviceaccount.com
bindings:
- members:
  - serviceAccount:ipi-min-perm-without-actAs-sa@openshift-qe.iam.gserviceaccount.com
  role: roles/iam.serviceAccountUser
etag: BwYjEASZQ6Y=
version: 1
$ 
```

Note: the step `ipi-deprovision-deprovision` may fail due to [OCPBUGS-42117 [GCP] destroying a cluster of IPI/UPI minimum permissions got error "Required 'compute.regionTargetTcpProxies.list' permission"](https://issues.redhat.com/browse/OCPBUGS-42117)

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-debug

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-gcp-ipi-mini-perm-marketplace-f28

**Comment by jianli-wei** [comment]
@jianlinliu please help review, thanks!

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/minimal-permission/without-actas/provision/cucushift-installer-rehearse-gcp-ipi-minimal-permission-without-actas-provision-chain.metadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+	"path": "cucushift/installer/rehearse/gcp/ipi/minimal-permission/without-actas/provision/cucushift-installer-rehearse-gcp-ipi-minimal-permission-without-actas-provision-chain.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/minimal-permission/without-actas/provision/cucushift-installer-rehearse-gcp-ipi-minimal-permission-without-actas-provision-chain.metadata.json:2]:
```suggestion
	"path": "cucushift/installer/rehearse/gcp/ipi/minimal-permission/without-actas/workers-marketplace/provision/cucushift-installer-rehearse-gcp-ipi-minimal-permission-without-actas-workers-marketplace-provision-chain.yaml",
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml [hunk 2]
```diff
@@ -2284,26 +2284,30 @@ tests:
     test:
     - chain: openshift-e2e-test-mco-qe-longduration
     workflow: cucushift-agent-vsphere-dualstack-install-sno
-- as: gcp-ipi-marketplace-f28
-  cron: 32 21 7 * *
+- as: gcp-ipi-mini-perm-marketplace-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly.yaml:2287]: We also need to update the existing upgrade ci together:
```
openshift-openshift-tests-private-release-4.17__amd64-nightly-4.17-upgrade-from-stable-4.17.yaml:- as: gcp-ipi-marketplace-f60
openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.17.yaml:- as: gcp-ipi-marketplace-f28
openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-upgrade-from-stable-4.18.yaml:- as: gcp-ipi-marketplace-f60
```

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**